src/DistributedMetadata/CatalogService.cpp

```diff
@@ -576,13 +605,26 @@ CatalogService::TableContainerPerNode CatalogService::buildCatalog(const NodePtr
 /// Merge a snapshot of tables from one host
 void CatalogService::mergeCatalog(const NodePtr & node, TableContainerPerNode snapshot)
 {
+    std::unique_lock guard{catalog_rwlock};
+
     if (snapshot.empty())
-    {
+    {   
+        /// Clear the table corresponding residual data
+        auto iter = indexed_by_node.find(node->identity);
+
+        if (iter != indexed_by_node.end())
+        {
+            for (const auto & p : iter->second)
+            {
+                indexed_by_id.erase(p.second->uuid);
+                indexed_by_name.erase(indexed_by_name.find(std::make_pair(p.second->database, p.second->name)));
```

Comment:
`indexed_by_name.erase(std::make_pair(p.second->database, p.second->name));` and please `assert` something is indeed erased

Reply:
fixed